integration/testrunner/db.go

```diff
@@ -24,37 +24,28 @@ import (
 )
 
 // queryIfAuditLogExists queries the DB and checks if audit log contained in the query exists or not.
-func queryIfAuditLogExists(ctx context.Context, tb testing.TB, query *bigquery.Query, expectedNum int64) (bool, error) {
+func queryIfAuditLogExists(ctx context.Context, tb testing.TB, query *bigquery.Query) ([]bigquery.Value, error) {
 	tb.Helper()
 
 	job, err := query.Run(ctx)
 	if err != nil {
-		return false, fmt.Errorf("failed to run query: %w", err)
+		return nil, fmt.Errorf("failed to run query: %w", err)
 	}
 
 	if status, err := job.Wait(ctx); err != nil {
-		return false, fmt.Errorf("failed to wait for query: %w", err)
+		return nil, fmt.Errorf("failed to wait for query: %w", err)
 	} else if err = status.Err(); err != nil {
-		return false, fmt.Errorf("query failed: %w", err)
+		return nil, fmt.Errorf("query failed: %w", err)
 	}
-
 	it, err := job.Read(ctx)
 	if err != nil {
-		return false, fmt.Errorf("failed to read job: %w", err)
+		return nil, fmt.Errorf("failed to read job: %w", err)
 	}
-
 	var row []bigquery.Value
 	if err = it.Next(&row); err != nil {
-		return false, fmt.Errorf("failed to get next row: %w", err)
+		return nil, fmt.Errorf("failed to get next row: %w", err)
 	}
-
-	// Check if the matching row count is equal to expected, if yes, then the audit log exists.
-	tb.Logf("Found %d matching rows", row[0])
```

Comment:
Do we still need to keep the check that matching row count is equal to expected and the check the diff?

Reply:
In addition, I checked the docs here https://pkg.go.dev/cloud.google.com/go/bigquery#RowIterator, I thought we need a for loop to figure out how many matched rows we found. Here, we may not need to iterate all the rows, if we found more than one row, we can fail the test.